### ISSUE TITLE
[AKS] User is unable to remove original node pool

### ISSUE THREAD

**Issue by timhaneunsoo** [rancher/dashboard#6071]
**Setup**
- Rancher version: 2.6-head 68c6c13 but also reproducible in 2.6.3
- Browser type & version:

**Describe the bug**

After provisioning an AKS cluster, user is unable to remove original node pool while editing cluster. There is an option to add and remove a new node pool, but none to remove the original node pool. 
![image](https://user-images.githubusercontent.com/97477402/171225596-bf98e7ee-2835-4ec7-afe2-93ce898377ca.png)

This action is available when editing an EKS cluster on the original node pool
![image](https://user-images.githubusercontent.com/97477402/171224799-53e10bb5-469d-4dc7-803f-8a16349e5edb.png)

**To Reproduce**

1. Provision AKS cluster
2. After cluster becomes active, edit cluster
3. Navigate down to `Node Pools` section
4. Verify that user can add node pool, and the new node pool can be removed
5. Verify that user cannot remove original node pool, even after new node pool is added

**Result**
User is not able to remove the original node pool.

**Expected Result**

User should be able to remove the original node pool, as they can in EKS

**Comment by mantis-toboggan-md** [comment]
This will be included in the work to port AKS provisioning from Ember to Vue: https://github.com/rancher/dashboard/issues/8965

One consequence of this change is that we need to add validation to ensure the user still has at least one pool with the system mode selected.

**Comment by nwmac** [comment]
@mantis-toboggan-md Zube has this in the 'To Test' column, but the label says Review - what's the status?

**Comment by mantis-toboggan-md** [comment]
@nwmac the correct status is 'To Test'

**Comment by valaparthvi** [comment]
Validated on Rancher(v2.9-a9355940c0629ca419b7dd3e4098c4ea0e52c0c0-head); Dashboard(master 46a44c1), it is fixed.
- I was able to delete the original nodepool and add a new one.
- I was able to see the error `At least one node pool must use the 'system' mode.` when I tried to remove all nodepool.

![Screenshot from 2024-07-10 14-50-43](https://github.com/rancher/dashboard/assets/11015077/0e36d762-ef59-4d4b-8d6b-6963427ea62a)
